common/src/main/java/org/apache/comet/vector/CometVector.java

```diff
@@ -89,6 +90,8 @@ public Decimal getDecimal(int i, int precision, int scale) {
       return createDecimal(getInt(i), precision, scale);
     } else if (!useDecimal128 && precision <= Decimal.MAX_LONG_DIGITS()) {
       return createDecimal(getLong(i), precision, scale);
+    } else if (useDecimal128 && precision <= Decimal.MAX_LONG_DIGITS()) {
+      return createDecimal(getLongFromDecimalBytes(getBinaryDecimal(i)), precision, scale);
```

Comment:
nit: for readability, I think it is better not to repeat the `precision <= Decimal.MAX_LONG_DIGITS()` expression

```suggestion
    } else if (precision <= Decimal.MAX_LONG_DIGITS()) {
      if (useDecimal128) {
        return createDecimal(getLongFromDecimalBytes(getBinaryDecimal(i)), precision, scale);
      } else {
        return createDecimal(getLong(i), precision, scale);
```